content/error-handling/error-handling-assignment/_index.md

```diff
@@ -0,0 +1,28 @@
+---
+content_type: project
+flavours:
+  - java
+  - javascript
+  - markdown
+  - python
+prerequisites:
+  hard:
+    - error-handling/javascript-error-handling
+    - error-handling/python-error-handling
+protect_main_branch: false
+ready: true
+submission_type: repo
+title: Error Handling Assignment
+---
+
+Please answer the following questions in markdown files and submit following the instructions below.
+
+## Questions
+
+1. In your programming  language, what is the fundamental purpose of error handling, and why is it crucial in software development?
+
+2. In your programming language, what are some poor practices in error handling that developers should avoid? Provide examples if possible.
+
+3. In your programming language, when should developers catch and handle errors, and what factors should influence this decision?
+
+4. In your programming language, demonstrate how to raise a custom error with a specific error message. Write a code snippet that showcases this, and explain why it's essential to include clear error messages in custom exceptions.
```

Comment:
- I don't think the "In your programming language" part is necessary for questions that are general and don't involve code.
- I think it would be good to have some code snippets related questions too and have the learner pick and explain certaint things like in the tdd written assignment.
- I think a question about why we use try catch/except when handling errors would be good because most learners don't know that it's for graceful handling of errors instead of breaking the whole program when it's not necessary
1. I think this should be one question, you can rephrase the it or remove the part from "and"
2. I think we should explicitly say if we want the learners to provide examples and mention how many(E.g. "provide at least 1 example")
3. I think this could be rephrased to one question or the last part of the question removed or made it's own question
- We should also link to the submission instructions for written assignments